internal/provider/resource_boundary_cluster.go

```diff
@@ -0,0 +1,260 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	boundarymodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-boundary-service/preview/2021-12-21/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+// defaultClusterTimeout is the amount of time that can elapse
+// before a cluster read operation should timeout.
+var defaultBoundaryClusterTimeout = time.Minute * 5
+
+// createUpdateBoundaryClusterTimeout is the amount of time that can elapse
+// before a cluster create operation should timeout.
+var createBoundaryClusterTimeout = time.Minute * 35
+
+// deleteBoundaryClusterTimeout is the amount of time that can elapse
+// before a cluster delete operation should timeout.
+var deleteBoundaryClusterTimeout = time.Minute * 25
+
+func resourceBoundaryCluster() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource allows you to manage an HCP Boundary cluster",
+		CreateContext: resourceBoundaryClusterCreate,
+		ReadContext:   resourceBoundaryClusterRead,
+		DeleteContext: resourceBoundaryClusterDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceBoundaryClusterImport,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create:  &createBoundaryClusterTimeout,
+			Delete:  &deleteBoundaryClusterTimeout,
+			Default: &defaultBoundaryClusterTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// required inputs
+			"cluster_id": {
+				Description:      "The ID of the Boundary cluster",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"username": {
+				Description:      "The username of the initial admin user. This must be at least 3 characters in length, alphanumeric, hyphen, or period.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateBoundaryUsername,
+			},
+			"password": {
+				Description:      "The password of the initial admin user. This must be at least 8 characters in length. Note that this may show up in logs, and it will be stored in the state file.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateBoundaryPassword,
+				Sensitive:        true,
+			},
+			"created_at": {
```

Comment:
✨ Just to help with quick parsing of the inputs
```suggestion
// computed outputs
			"created_at": {
```

Reply:
Updated! Good catch!